plugins/casutg/src/main/java/com/blazemeter/jmeter/threads/concurrency/ConcurrencyThreadStarter.java

```diff
@@ -53,14 +70,32 @@ private long getPlannedConcurrency() {
         }
 
         if (rampUp == 0 || timeOffset > rampUp) {
-            return Math.round(maxConcurr);
+            return maxConcurr;
         } else if (steps > 0) {
             double stepSize = maxConcurr / (double) steps;
             double stepLen = rampUp / (double) steps;
             return Math.round(stepSize * (Math.floor(timeOffset / stepLen) + 1));
         } else {
-            double slope = maxConcurr / rampUp;
+            double slope = (double) maxConcurr / rampUp;
             return Math.round(slope * timeOffset);
         }
     }
+
+    void checkNeedsPropertiesReloading(long now) {
+        if (CACHING_VALIDITY_MS > 0 && now - lastCachedTime > CACHING_VALIDITY_MS) {
+            this.rampUp = owner.getRampUpSeconds();
+            this.hold = owner.getHoldSeconds();
+            this.steps = owner.getStepsAsLong();
+            this.maxConcurr = Math.round(owner.getTargetLevelAsDouble());
```

Comment:
original algorighm has no rounding. I'd avoid introducing it unless there is good reason.

Reply:
getTargetLevelAsDouble() transforms an int into a double. I do it once to avoid redoing it later each time

Reply:
This can change the math afterwards. Can we keep the math, unless we have a bug there?
These changes affect so wide audience that I want to avoid changes as much as possible.

Reply:
In fact I do the same as before but I cache. Do you want me to revert ?

Reply:
Yes, please